system-addon/common/Reducers.jsm

```diff
@@ -265,6 +265,22 @@ function Sections(prevState = INITIAL_STATE.Sections, action) {
           return item;
         })
       }));
+    case at.PLACES_SAVED_TO_POCKET:
+      if (!action.data) {
+        return prevState;
+      }
+      return prevState.map(section => Object.assign({}, section, {
+        rows: section.rows.map(item => {
+          if (item.url === action.data.url) {
```

Comment:
There's a slight bug / accidental(?) fix here in that if you pocket a story, stories' `rows` gets deduped via `SECTION_UPDATE`. A similar thing happens when bookmarking that @piatra and @csadilek worked on.

The bug is that even when Highlights no longer has the deduping url, e.g., archived pocket or unbookmarked page+delete history, `rows` is still deduped until it gets refreshed with an update or restart.

Simplest steps to reproduce:
1) save to pocket a story
2) archive that card
3) refresh page see that the card is still gone
4) toggle stories section from preferences, and see the story come back

Probably not really worth a fix as the user wanted it gone anyway and would most likely only run into it on restarting firefox, and by then, hopefully new stories would have come in…

Reply:
Ah I see, yeah that's unfortunate but I think you're right, not super important to fix. Unless we added the saved to pocket trending card to the block list when they archive or save it or something. But probably not worth fixing right now